scipio/examples/hello_world.rs

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the
+// MIT/Apache-2.0 License, at your convenience
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2020 Datadog, Inc.
+//
+use futures::future::join_all;
+use scipio::{Local, LocalExecutor};
+
+async fn hello() {
+    let mut tasks = vec![];
+    for t in 0..5 {
+        tasks.push(Local::local(async move {
+            println!("{}: Hello {} ...", Local::id(), t);
+            Local::later().await;
+            println!("{}: ... {} World!", Local::id(), t);
+        }));
+    }
+    join_all(tasks).await;
+}
+
+fn main() {
+    // There are two ways to create an executor, demonstrated in this example.
+    //
+    // We can create it in the current thread, and run it separately later...
+    let ex = LocalExecutor::new(Some(0)).unwrap();
+
+    // Or we can spawn a new thread with an executor inside.
+    LocalExecutor::spawn_new("hello", Some(1), async move {
```

Comment:
The _new_ in `spawn_new` looks odd because it's implied by the verb. How about just `spawn`? Or `spawn_task`, `spawn_thread`?

Reply:
Naming is unfortunate here. There is already a `spawn` that pushes a fiber on existing executor:

```rust
    pub fn spawn<T: 'static>(&self, future: impl Future<Output = T> + 'static) -> Task<T> {
```

Reply:
I agree `spawn_new` is not ideal but there are many other executor related functions called `spawn` and its variants.
I don't like `spawn_task` because this spawns an executor, not a a task.

How about we just go full verbose then and call it `spawn_executor` ?

Reply:
`spawn_executor` seems fine to me.